### PR TITLE
Stop the size cache checking loop

### PR DESCRIPTION
**Pull request by Florian1209**

### CODE REVIEW

**Comment by marpet** [comment]
Actually I'm still reviewing. But I will send a pull request.

**Comment by Florian1209** [comment]
@marpet 
Damn! There is issue on SNAP closing.
The application has blocked after the below logging
INFO [org.esa.snap.rcp.SnapApp]: Request to stop SNAP Desktop
INFO [org.esa.snap.rcp.SnapApp]: Stopping SNAP Desktop

**Comment by marpet** [comment]
I've send a PR. Would be good if you could test it. I only implemented it but haven't run it.
https://github.com/senbox-org/s2tbx/pull/93

File: s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java [hunk 1]
```diff
@@ -63,7 +66,16 @@ public void process() throws Exception {
...
                 executor.execute(runnable);
             } catch (Exception ex) {
-                logger.log(Level.SEVERE, "Failed to initialize S2Cache options."+ex.getMessage());
+                logger.log(Level.SEVERE, "Failed to initialize S2Cache options.",ex);
             }
     }
+
+    public void complete() {
+        try {
+            executor.complete();
```
Comment (lnicola) [inline s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java:75]: I don't think this does anything. `ThreadExecutor.complete()` calls `ThreadPoolExecutor.shutdown()`, which not only doesn't interrupt the `Thread.sleep` call, but also doesn't kill the running tasks:

> Initiates an orderly shutdown in which previously submitted tasks are executed, but no new tasks will be accepted.

Generally, killing running threads is a finicky operations, hence my suggestion to use a timer.

Comment (marpet) [reply s2tbx-cache/src/main/java/org/esa/s2tbx/dataio/cache/S2CacheSizeChecking.java:75]: Yes, one one my reasons for the PR

**Pull request merged**